uberpoet/cpprojectgen.py

```diff
@@ -75,27 +79,59 @@ def example_command(self):
 
     # Generation Functions
 
-    def gen_app(self, app_node, node_list, target_loc, loc_json_file_path):
+    def gen_app(self, app_node, node_list, target_swift_loc, target_objc_loc, loc_json_file_path):
         library_node_list = [n for n in node_list if n.node_type == ModuleNode.LIBRARY]
 
         if loc_json_file_path:
             loc_reader = locreader.LocFileReader()
             loc_reader.read_loc_file(loc_json_file_path)
-            loc = loc_reader.loc_for_module(n.name)
```

Comment:
this was the severe bug I mentioned above. Apparently the `n` variable is the last entry from the loop above on line 79. I am beginner/intermediate in python so I had no idea.

We now correctly get the LOC for each module before we generate it.